doc/dynamic-environments/configuration.mkd

```diff
@@ -454,3 +477,46 @@ This will create the following directory structure:
 |-- app1_production  # app1 data repository, production branch
 |-- app1_develop     # app1 data repository, develop branch
 ```
+
+### Filtering branches
+
+#### Filtering branches - Exclude branches with a prefix
+
+If you would like to filter out all branches starting with and including "dev-".
```

Comment:
Excluding branches this way feels a bit awkward. I wonder if having two filters, one for include and one for exclude would be better? That way you could write positive regexes in both cases, which is usually easier. I think in that case you would add `'^dev-.*'` to an exclude regex and any matches would be excluded. Not sure if that is better or not.

Reply:
What happens if they specify both filters, then which filter applies?

I would be more inclined to implement an additional option branch_filter_type that can either be include or exclude. Then the regex will be interpreted as either inclusive or exclusive.

Reply:
I'm a big fan of the simplicity of having one filter and using the regex negation. I expect that users getting to this level of usage will be able to follow the examples or be able to look it up quickly.

Reply:
I feel like adding a setting that is a regex is probably a sign that we're taking the wrong approach to solving the problem. Do users actually need full regex matching, or would a simple allow/reject on a prefix or lists of allowed/denied environments be adequate?
Regexes tend to be simple in implementation, but not in actual usage, and debugging them can be challenging.

Reply:
I can see users wanting to both be able to use a list of denied/allowd environments, and a list of denied/allowed prefixes. As I start thinking about how to document how this works for users, it gets ...uh... complicated. Do you have an configuration spec in mind?

Reply:
I would probably start with something like `ignore_branches` or `ignore_branch_prefixes` as a yaml array. I suspect that having a way to reject environments with a given prefix would be sufficient for most users.

Reply:
`ignore_branch_prefixes` could also be a list of environments to ignore.

Reply:
I like it.